renku/core/models/workflow/plan.py

```diff
@@ -44,13 +44,26 @@
 class Plan:
     """Represent a `renku run` execution template."""
 
-    def __init__(self, id_, arguments=None, command=None, inputs=None, name=None, outputs=None, success_codes=None):
+    def __init__(
+        self,
+        id_,
+        arguments=None,
+        command=None,
+        description=None,
+        inputs=None,
+        keywords=None,
+        name=None,
+        outputs=None,
+        success_codes=None,
+    ):
         """Initialize."""
         self.arguments = arguments or []
         self.command = command
+        self.description = description
         self.id_ = id_
         self.inputs = inputs or []
-        self.name = name or "{}-{}".format(secure_filename(self.command), uuid.uuid4().hex)
+        self.keywords = keywords or []
+        self.name = name or f"{secure_filename(self.command)}-{uuid.uuid4().hex}"
```

Comment:
This will result in many "python-abcdefg" runs. I wonder if we should use self.to_argv() and limit it to 15-20 characters, so that it's a bit more descriptive, at least potentially.

Reply:
This is a bit challenging to be done the right because we want the name to be unique as well while we don't load the whole DependencyGraph. For now, I've modified the `Plan.from_run` to pass a name to this constructor according to your suggestion. This constructor is not called directly.